pkg/catalogv2/oci/client.go

```diff
@@ -34,6 +34,15 @@ import (
 // maxHelmChartTar defines what is the max size of helm chart we support.
 const maxHelmChartTarSize int64 = 20 * 1024 * 1024 // 20 MiB
 
+// Default Values for exponentialBackOff function which is used
+// by oras to retry a HTTP call when 429 response code is hit.
+var retryPolicy retry.GenericPolicy = retry.GenericPolicy{
+	Retryable: retry.DefaultPredicate,
+	MinWait:   1 * time.Second,
+	MaxWait:   5 * time.Second,
+	MaxRetry:  5,
+}
```

Comment:
When accessing a rate-limited server under distress we probably want to way longer before retrying. If I read this correctly, we would do our 6 requests in 20 seconds.

It is also worth checking the [Retry-After](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Retry-After) header. If set, it can help us avoid prolonging the rate-limited state by not retrying when asked not to for a given period of time.

Reply:
One limitation here is that the retry is handled by the oras library, and afaik, we can't set one time for 429 and another for other errors. Regarding the header, oras handles it [here](https://github.com/oras-project/oras-go/blob/11d464f8432e77175bc9cda221ec2d797eac752c/registry/remote/retry/policy.go#L95)

Reply:
Also @pjbgf this is configurable.